### PR TITLE
test: 100% Coverage of SelectCurrency page inside setup account

### PR DESCRIPTION
**Pull request by suyashpatil78**
https://app.clickup.com/t/85zt052f3

Screenshot 2023-05-30 at 11 12 25 AM

### CODE REVIEW

File: src/app/post-verification/setup-account/select-currency/select-currency.component.html [hunk 1]
```diff
@@ -11,7 +11,7 @@
       />
     </mat-form-field>
     <ion-buttons slot="end">
-      <ion-button (click)="onDoneClick()"> Done </ion-button>
+      <ion-button data-testid="doneBtn" (click)="onDoneClick()"> Done </ion-button>
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/select-currency/select-currency.component.html:14]: We can directly get the ion-button from css. No need to add test id

File: src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts [hunk 2]
```diff
@@ -1,26 +1,114 @@
...
+    loaderService.showLoader.and.resolveTo();
+    loaderService.hideLoader.and.resolveTo();
+
+    fixture.detectChanges();
+    component.ngOnInit();
+    tick(1000);
+    expect(loaderService.showLoader).toHaveBeenCalled();
+    expect(currencyService.getAll).toHaveBeenCalled();
+    expect(loaderService.hideLoader).toHaveBeenCalled();
+    expect(component.currencies$).toBeDefined();
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:58]: Compare with expected value

File: src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts [hunk 3]
```diff
@@ -1,26 +1,114 @@
...
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:73]: same as above

File: src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts [hunk 4]
```diff
@@ -1,26 +1,114 @@
...
+  }));
+
+  it('should load currencies on initialization even if value is empty', fakeAsync(() => {
```
Comment (yksolanki9) [inline src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:61]: What's the difference between this and previous test case?

**Review by yksolanki9 — CHANGES_REQUESTED** [review]
Minor comments

File: src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts (hunk 4 above)
Comment (suyashpatil78) [reply src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:61]: This test is to cover the branch `|| shortCode`, if there is not value corresponding to the shortcode, then the `longName` takes the value of `shortcode`

Comment (suyashpatil78) [reply src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:61]: I have combined the two tests into a single test to check both the branches

File: src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts [hunk 5]
```diff
@@ -1,26 +1,113 @@
...
+    loaderService.showLoader.and.resolveTo();
+    loaderService.hideLoader.and.resolveTo();
+
+    fixture.detectChanges();
+    tick(500);
+    expect(loaderService.showLoader).toHaveBeenCalledTimes(1);
+    expect(currencyService.getAll).toHaveBeenCalledTimes(1);
+    expect(loaderService.hideLoader).toHaveBeenCalledTimes(1);
+    component.currencies$.subscribe((currencies) => {
+      expect(currencies).toEqual([
```
Comment (Dimple16) [inline src/app/post-verification/setup-account/select-currency/select-currency.component.spec.ts:58]: move this to mock data file

**Review by Dimple16 — APPROVED** [review]
minor comments

**Review by yksolanki9 — APPROVED** [review]

**Pull request merged**